src/DemoFile/DemoFileReader.Parallel.cs

```diff
@@ -91,10 +95,10 @@ await reader.MoveNextCoreAsync(cmd.Command, cmd.IsCompressed, cmd.Size, cancella
             {
             }
 
-            return new[] {result};
+            return new[] {initialResult, result};
```

Comment:
it seems that `initialResult` is missing

Reply:
I couldn't find which test needs to be modified. Do you want me to add new test for this case ?

Reply:
Yes please - once that's done this PR is good to merge